src/main/java/seedu/address/logic/commands/AnswerCommand.java

```diff
@@ -34,18 +39,23 @@ public AnswerCommand(Answer attemptedAnswer) {
     @Override
     public CommandResult execute(Model model, CommandHistory history) throws CommandException {
         requireNonNull(model);
+        List<Card> lastShownList = model.getFilteredCards();
+
         if (!model.checkIfInsideTestSession()) {
             throw new CommandException(Messages.MESSAGE_INVALID_COMMAND_OUTSIDE_TEST_SESSION);
         }
         if (model.checkIfCardAlreadyAnswered()) {
             throw new CommandException(Messages.MESSAGE_INVALID_ANSWER_COMMAND);
         }
         model.setCardAsAnswered();
+        Card cardToMark = lastShownList.get(0);
```

Comment:
Actually, should we use getCurrentTestedCard method from model to get the cardToMark? :)

Reply:
You're right, I changed it. Thx!